### PR TITLE
feat(DSTEW-2005): Added Area-of-law based claim field amendability validation gate

### PR DESCRIPTION
**Pull request by vvartakMoJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/DSTEW-2005)

Implements the field amendability gate from the parent ticket (DSTEW-1593). For every provider-requested changed field, the gate checks whether that field is amendable for the claim's Area of Law and collects a field-level validation error for any field that is not.
## How

- AmendableClaimFields — a new, auditable registry of the amendable claim fields, organised per Area of Law (CRIME_LOWER, MEDIATION, LEGAL_HELP) as three flat whitelists behind an AreaOfLaw → Set<fieldIdentifier> map. Structured to mirror the signed-off AaBC artefact "Amend MVP - fields for amendment" one line at a time for reviewability. Exposes isAmendable(field, areaOfLaw).
- FieldAmendabilityValidationStep — new validation step. Reads the claim's Area of Law from the before-state, filters the assembled diff to REQUESTED changes, and emits INVALID_FIELD_NOT_AMENDABLE_FOR_AREA_OF_LAW (severity ERROR, non-fatal) for each non-amendable field, naming the field in the message so the failure is attributed to it.
- ClaimAmendmentValidationCode — new INVALID_FIELD_NOT_AMENDABLE_FOR_AREA_OF_LAW code.
- ClaimAmendmentValidationService — registers the step in STEP_ORDER at the amendability slot (after assessed-pricing, before the external steps), so its findings aggregate into the Step 12 outcome alongside other steps.
## Behaviour

- Non-amendable changed field → error collected and attributed to that field.
- Non-fatal, so it aggregates with other validation messages rather than short-circuiting.
- Absent/unknown Area of Law → every changed field is treated as non-amendable.
- FSP-sourced (consequential) changes are out of scope; only provider-requested edits are gated.
- The step is side-effect-free; any resulting error prevents the amendment being applied (no claim update, amendment record, calculated-fee row or event), consistent with the existing pipeline.

## Tests

- AmendableClaimFieldsTest — per-area membership consistency, representative artefact rows, and edge cases (null field, null area, unregistered field).
- FieldAmendabilityValidationStepTest — amendable vs non-amendable outcomes, correct code/attribution, non-fatal severity, mixed changed fields (only non-amendable reported), unregistered field, FSP-change exclusion, empty diff, and missing Area of Law.
- ClaimAmendmentValidationServiceIntegrationTest — assembled-chain test proving a non-amendable field aggregates with another error through the real wiring (Step 12).

## Checklist

Before you ask people to review this PR please ensure the following and mark as complete when done:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.
- [ ] If applicable, you have executed the end-to-end (E2E) tests using the [bulk-submission-and-fee-scheme-tests-](https://github.com/ministryofjustice/bulk-submission-and-fee-scheme-tests-) repository and confirmed they pass.

### CODE REVIEW

**Review by BenJusticeDigital — APPROVED** [review]
LGTM, not sure if you want to address coPilot, seems minor

File: claims-data/service/src/integrationTest/java/uk/gov/justice/laa/dstew/payments/claimsdata/service/amendment/ClaimAmendmentValidationServiceIntegrationTest.java [hunk 1]
```diff
@@ -251,6 +252,57 @@ void emptyReferenceDataReturnsFatalTechnicalError() {
...
+    @DisplayName("a non-amendable changed field is collected and aggregates with other errors")
+    void nonAmendableFieldAggregatesWithOtherErrors() {
+      ClaimAmendmentState state = stateWithNonAmendableChange();
+
+      assertThat(validationService.validateAmendmentRequest(state))
+          .extracting(ClaimAmendmentValidationError::getCode)
+          .contains(
+              ClaimAmendmentValidationCode.INVALID_FIELD_NOT_AMENDABLE_FOR_AREA_OF_LAW.toString(),
+              ClaimAmendmentValidationCode.INVALID_USER_IDENTIFIER_FORMAT.toString());
+    }
```
[the remark this answers is not in the document]
Comment (vvartakMoJ) [reply claims-data/service/src/integrationTest/java/uk/gov/justice/laa/dstew/payments/claimsdata/service/amendment/ClaimAmendmentValidationServiceIntegrationTest.java:269]: Additional errors are not the main concern of this change, so ignoring this comment.
